mycore-user2/src/main/java/org/mycore/user2/login/MCRLDAPClient.java

```diff
@@ -258,8 +221,164 @@ private String formatName(String name) {
         return trimmedName.substring(pos + 1) + ", " + trimmedName.substring(0, pos);
     }
 
+    public enum SecurityAuthentication {
+
+        NONE,
+
+        SIMPLE;
+
+    }
+
+    public record ConnectionSettings(
+        String providerUrl,
+        SecurityAuthentication securityAuthentication,
+        String securityProtocol,
+        String securityPrincipal,
+        String securityCredentials,
+        Integer connectTimeoutMillis,
+        Integer readTimeoutMillis) {
+
+        public ConnectionSettings validate() {
+            Objects.requireNonNull(providerUrl, "Provider URL must not be null");
+            Objects.requireNonNull(securityAuthentication, "Security authentication must not be null");
+            Objects.requireNonNull(securityProtocol, "Security protocol must not be null");
+            if (securityAuthentication == SecurityAuthentication.SIMPLE) {
+                Objects.requireNonNull(securityPrincipal, "Security principal must not be null");
+                Objects.requireNonNull(securityCredentials, "Security credentials must not be null");
+            } else if (securityPrincipal != null || securityCredentials != null) {
+                LOGGER.warn("Security principal or credentials given, even though the authentication is 'NONE'");
+            }
+            Objects.requireNonNull(connectTimeoutMillis, "Read timeout must not be null");
+            Objects.requireNonNull(readTimeoutMillis, "Read timeout must not be null");
+            return this;
+        }
+
+        public Map<String, String> toLdapSettings() {
```

Comment:
Only values that are not null should be included in `ldapEnv`. Otherwise line 114 leads to a NPE when creating the `Hashtable` with null values.

Reply:
Good catch. Will change.